chain/chain/src/chain.rs

```diff
@@ -2949,9 +2947,6 @@ impl Chain {
             .get_block(block_header.prev_hash())
         {
             Ok(prev_block) => {
-                if shard_id as usize >= prev_block.chunks().len() {
```

Comment:
Note to reviewer: I am not 100% convinced that this is correct so please double check that this is fine.

Reply:
Seems correct to me.
I'm more worried about using the same `shard_id` value for two different shard layouts: prev epoch and the `sync_hash` epoch.
Fixing that is out of scope of your PR.